compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -450,9 +450,23 @@ impl<'a> Resolver<'a> {
                 // let foo =...
                 //     ^^^ given this Span
                 // ------- get this Span to have an applicable suggestion
+
+                // edit:
+                // only do this if the const and usage of the non-constant value are on the same line
+                // the further the two are apart, the higher the chance of the suggestion being wrong
+                // also make sure that this line isn't the first one (ICE #90878)
+
                 let sp =
                     self.session.source_map().span_extend_to_prev_str(ident.span, current, true);
-                if sp.lo().0 == 0 {
+
+                let is_first_line = self
+                    .session
+                    .source_map()
+                    .lookup_line(sp.lo())
+                    .map(|file_and_line| file_and_line.line == 0)
+                    .unwrap_or(true);
+
+                if sp.lo().0 == 0 || self.session.source_map().is_multiline(sp) || is_first_line {
```

Comment:
You can avoid the ICE by checking for overflow here, this removes the need to check the line number:

```suggestion
                if sp.lo().0.saturating_sub(current.len() as u32) == 0 || self.session.source_map().is_multiline(sp) {
```

Reply:
That is certainly nicer and better, thank you!